File: seohwee/week3/A034.cpp
#include <iostream>

using namespace std;

int main(void){
    int num;
    int count[42]={0};
    int result=0;

    for(int i=0; i<10; i++){
        cin >> num;
        count[num % 42]++;
    } 
    
    for(int num : count){
        if(num > 0) result++;
    }
    cout << result;
}

File: SeongKyungKim/week4/A034.java
import java.util.Scanner;

public class Main{
	public static void main(String args[]) {
		int[] remainder = new int[10];
		int[] count = new int[42]; //0~41
		int result=0;
		Scanner sc = new Scanner(System.in);
		
	    for(int i=0; i<10; i++) {
	    	remainder[i] = (sc.nextInt())%42;
	    	//System.out.println("remainder["+i+"]: " + remainder[i]);
	        count[remainder[i]] ++;
	    }
	    
	    for(int i=0; i<42; i++) {
	    //	System.out.println(count[i]);
	    	if(count[i]>1) result+=count[i]-1;
	    }
	    System.out.println(10-result);
	    sc.close();
	}
}
